petab/parameters.py

```diff
@@ -26,22 +27,49 @@ def get_parameter_df(
     if parameter_file is None:
         return parameter_file
 
+    parameter_df = None
+
+    if isinstance(parameter_file, pd.DataFrame):
+        parameter_df = parameter_file
+
     if isinstance(parameter_file, str):
-        parameter_file = pd.read_csv(parameter_file, sep='\t')
+        parameter_df = pd.read_csv(parameter_file, sep='\t')
+
+    if isinstance(parameter_file, list):
+        for parameter_subset_file in parameter_file:
+            parameter_df = pd.concat([parameter_df, pd.read_csv(
+                parameter_subset_file, sep='\t')])
+            # Remove identical parameter definitions
+            parameter_df.drop_duplicates(inplace=True, ignore_index=True)
+            # Check for contradicting parameter definitions
+            if (parameter_df.nunique()[PARAMETER_ID]
+                    < parameter_df.count()[PARAMETER_ID]):
+                parameter_duplicates = set(
+                    parameter_df[PARAMETER_ID].loc[parameter_df.duplicated(
+                        subset=[PARAMETER_ID])]
+                    )
+                raise ValueError(
+                    f'The values of {PARAMETER_ID} must be unique or'
+                    + ' identical between all parameter subset files. The'
+                    + ' following duplicates were found:\n'
+                    + '{}\n'.format('\n'.join(parameter_duplicates))
+                    + 'The following file contained at least one copy of'
+                    + ' these duplicates:\n' + f'"{parameter_subset_file}".'
```

Comment:
```suggestion
                    f'The values of {PARAMETER_ID} must be unique or'
                    ' identical between all parameter subset files. The'
                    ' following duplicates were found:\n'
                    f'{parameter_duplicates}'
                    'The following file contained at least one copy of'
                    f' these duplicates: {parameter_subset_file}.'
```

Reply:
Good idea, duplicates are now reported as a set that could be copied into a python script to identify files containing the duplicated `parameterId`'s (although, such a feature might be better suited for `petablint`).